include/net/mqtt.h

```diff
@@ -715,6 +715,21 @@ int mqtt_read_publish_payload(struct mqtt_client *client, void *buffer,
 int mqtt_read_publish_payload_blocking(struct mqtt_client *client, void *buffer,
 				       size_t length);
 
+/**
+ * @brief Blocking version of @ref mqtt_read_publish_payload function which
+ *        runs until the required number of bytes are read.
+ *
+ * @param[in] client Client instance for which the procedure is requested.
+ *                   Shall not be NULL.
+ * @param[out] buffer Buffer where payload should be stored.
```

Comment:
Btw, @jukkar, @tbursztyka, I always forget this doxygen syntax for in/out params, IMHO we should adopt it for the rest of networking API, because in many places, "out" params either not clear, or quite mouthful.

But OTOH, I don't see any in/out indication in the rendered docs, e.g.: https://docs.zephyrproject.org/latest/reference/networking/mqtt.html?highlight=mqtt#_CPPv325mqtt_read_publish_payloadP11mqtt_clientPv6size_t

So, q to @rlubos: are you sure that's the correct syntax to use? And to @dbkinder: can we do anything about it?

But even if it's only for .h files, it's useful helper IMHO.

Reply:
Yes it is, see http://www.doxygen.nl/manual/commands.html#cmdparam

I guess that visibility of this attribute depends on doxygen configuration? It is present in our nRF5 SDK API documentation.